sdk/storage/azure-storage-blob/azure/storage/blob/_shared/response_handlers.py

```diff
@@ -60,9 +61,13 @@ def normalize_headers(headers):
 
 def deserialize_metadata(response, obj, headers):  # pylint: disable=unused-argument
     try:
-        raw_metadata = {k: v for k, v in response.http_response.headers.items() if k.startswith("x-ms-meta-")}
+        raw_metadata = {
+            k: v for k, v in response.http_response.headers.items() if re.search("x-ms-meta-", k, re.IGNORECASE)
```

Comment:
We should use `re.compile` outside of the loops.

Reply:
Yessir, and we should also make the pattern a raw string to be consistent with best Python practices.